x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/NlpInferenceInput.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.inference.deployment;
+
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
+import org.elasticsearch.xpack.core.ml.inference.TrainedModelInput;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+
+import java.util.Map;
+
+/**
+ * Input is either a string or a document map
+ */
+public class NlpInferenceInput {
+
+    private final String inputText;
+    private final Map<String, Object> doc;
+
+    public NlpInferenceInput(String inputText) {
+        this.inputText = ExceptionsHelper.requireNonNull(inputText, "input_text");
+        doc = null;
+    }
+
+    public NlpInferenceInput(Map<String, Object> doc) {
+        this.doc = ExceptionsHelper.requireNonNull(doc, "doc");
+        this.inputText = null;
+    }
```

Comment:
purely preference, it would be great if these were instead `static` methods and a single private constructor

like:
```
fromText(....)
```
or 
```
fromDoc(...)
```

Reply:
Yup that's better